src/main/java/io/github/thebusybiscuit/slimefun4/utils/SlimefunUtils.java

```diff
@@ -319,17 +321,23 @@ private static boolean equalsItemMeta(@Nonnull ItemMeta itemMeta, @Nonnull ItemM
             return false;
         } else if (itemMeta.hasDisplayName() && displayName.isPresent() && !itemMeta.getDisplayName().equals(displayName.get())) {
             return false;
-        } else if (!checkLore) {
-            return true;
-        } else {
+        } else if (checkLore) {
             Optional<List<String>> itemLore = meta.getLore();
 
-            if (itemMeta.hasLore() && itemLore.isPresent()) {
-                return equalsLore(itemMeta.getLore(), itemLore.get());
-            } else {
-                return !itemMeta.hasLore() && !itemLore.isPresent();
+            if (itemMeta.hasLore() && itemLore.isPresent() && !equalsLore(itemMeta.getLore(), itemLore.get())) {
+                return false;
+            } else if (itemMeta.hasLore() != itemLore.isPresent()) {
+                return false;
             }
         }
+
+        // Fixes #3133: name and lore are not enough
+        OptionalInt itemCustomModelData = meta.getCustomModelData();
```

Comment:
While we are at it, `meta` should be renamed to `snapshot`, now that the class was renamed.

Reply:
> Just one variable name is a bit confusing (not your fault).
Which variable and what would you like the name to be?
I can rename meta easily.

As for CustomModelData, the equalsItemMeta method I have modified is only ever called inside isItemSimilar, and only after a check for the IDs has been performed (lines 281-285 or, due to the cargo filter issue with wrappers, 290-300) - I double checked :)